### PR TITLE
Support GHC 8.8

### PR DESCRIPTION
**Pull request by sjakobi**
This is blocked on:

* [x] https://github.com/timbod7/haskell-chart/pull/186

### CODE REVIEW

**Comment by sjakobi** [comment]
BTW is this the right repo at all?

https://github.com/bgamari/criterion-compare/blob/2f869b1375bae2192c502b753080aeeb3c5a560d/criterion-compare.cabal#L38-L40

says otherwise…

**Comment by sjakobi** [comment]
@bgamari, @AndreasPK This builds fine for me now.

Please let me know if I should submit this to Andreas' repo instead.

**Comment by AndreasPK** [comment]
@sjakobi This repo should be fine.

**Comment by sjakobi** [comment]
@AndreasPK Should I change the `source-repository` in the Cabal file then?

**Comment by bgamari** [comment]
Yes, changing the source repository would be appreciated. Thanks @sjakobi!

**Review by bgamari — APPROVED** [review]

File: criterion-compare.cabal [hunk 1]
```diff
@@ -19,12 +19,12 @@ executable criterion-compare
   hs-source-dirs:      src
   other-extensions:    GeneralizedNewtypeDeriving, RecordWildCards, OverloadedStrings, FlexibleContexts, TupleSections
   default-language:    Haskell2010
-  build-depends:       base >=4.9 && <= 4.13,
+  build-depends:       base >=4.9 && <= 4.14,
```
Comment (sjakobi) [inline criterion-compare.cabal]: I noticed that I can already build cleanly with 8.10.1-alpha2…

```suggestion
  build-depends:       base >=4.9 && <= 4.15,
```

Comment (phadej) [reply criterion-compare.cabal]: Don't use `<=` (less than or **equal to**). It's definitely not right, it should be `< 4.15` (less than)

File: criterion-compare.cabal [hunk 2]
```diff
@@ -19,12 +19,12 @@ executable criterion-compare
...
                        cassava >=0.5 && <0.6,
                        containers >=0.6 && <0.7,
                        Chart >=1.6 && < 1.10,
                        data-default >=0.5 && < 7.2,
-                       lens >=4.13  && < 4.18,
+                       lens >=4.13  && < 4.19,
```
Comment (phadej) [inline criterion-compare.cabal]: and this is `<4.20` now...

File: criterion-compare.cabal (hunk 1 above)
Comment (sjakobi) [reply criterion-compare.cabal]: Oops! ;)

**Comment by sjakobi** [comment]
Thanks for your comments @phadej! I updated the bounds. The diagrams ecosystem still needs updating so I've tried building with `--allow-newer`. `cassava` failed to build then:

```
Building library for cassava-0.5.2.0..

src/Data/Csv/Conversion.hs:26:3: error:
     error: #error **INVARIANT BROKEN** Detected invalid combination of `text-short` and `bytestring` versions. Please verify the `pre-bytestring-0.10-4` flag-logic in the .cabal file wasn't elided.
     # error **INVARIANT BROKEN** Detected invalid combination of `text-short` and `bytestring` versions. Please verify the `pre-bytestring-0.10-4` flag-logic in the .cabal file wasn't elided.
       ^
   |
26 | # error **INVARIANT BROKEN** Detected invalid combination of `text-short` and `bytestring` versions. Please verify the `pre-bytestring-0.10-4` flag-logic in the .cabal file wasn't elided.
   |   ^

src/Data/Csv/Conversion.hs:100:0: error:
     warning: "MIN_VERSION_text_short" is not defined [-Wundef]
     #if MIN_VERSION_text_short(0,1,0)
     ^
    |
100 | #if MIN_VERSION_text_short(0,1,0)
    | ^

src/Data/Csv/Conversion.hs:100:0: error:
     error: missing binary operator before token "("
    |
100 | #if MIN_VERSION_text_short(0,1,0)
    | ^

src/Data/Csv/Conversion.hs:1031:0: error:
     warning: "MIN_VERSION_text_short" is not defined [-Wundef]
     #if MIN_VERSION_text_short(0,1,0)
     ^
     |
1031 | #if MIN_VERSION_text_short(0,1,0)
     | ^

src/Data/Csv/Conversion.hs:1031:0: error:
     error: missing binary operator before token "("
     |
1031 | #if MIN_VERSION_text_short(0,1,0)
     | ^

src/Data/Csv/Conversion.hs:1:1: error:
    `cc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE
  | ^
```

Not sure what that's about…

**Comment by phadej** [comment]
don't use blunt `--allow-newer`. Use more specific `--allow-newer=base` or better `--allow-newer=diagrams-lib:base`. If you relax all bounds, it's unlikely anything will work.

**Comment by sjakobi** [comment]
> Use more specific `--allow-newer=base`

Thanks, that worked! I somehow hadn't been aware of this syntax – it's great! :)

**Pull request merged**